### PR TITLE
Added Italian translation to docs file index.md

### PR DESCRIPTION
**Pull request by csr**
This PR translates the index.md to Italian. This page was already present in the mkdocs.yml file.

### CODE REVIEW

File: docs/it/docs/index.md [hunk 1]
```diff
@@ -196,54 +191,54 @@ INFO:     Application startup complete.
...
-And now, go to <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.
+Adesso accedi su <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.
 
-You will see the alternative automatic documentation (provided by <a href="https://github.com/Rebilly/ReDoc" class="external-link" target="_blank">ReDoc</a>):
+Vedrai la documentazione interattiva dell'API (offerta da <a href="https://github.com/Rebilly/ReDoc" class="external-link" target="_blank">ReDoc</a>):
 
 ![ReDoc](https://fastapi.tiangolo.com/img/index/index-02-redoc-simple.png)
 
-## Example upgrade
+## Esempio più avanzato
```
Comment (pieromacaluso) [inline docs/it/docs/index.md:237]: Maybe "Esempio di aggiornamento" could be better for the context.

File: docs/it/docs/index.md [hunk 2]
```diff
@@ -273,175 +268,175 @@ def update_item(item_id: int, item: Item):
...
 ![ReDoc](https://fastapi.tiangolo.com/img/index/index-06-redoc-02.png)
 
-### Recap
+### Ricapitolando
 
-In summary, you declare **once** the types of parameters, body, etc. as function parameters. 
+Ricapitolando, è sufficiente dichiarare **una sola volta** i tipi dei parametri, del body, ecc. come parametri di funzioni.
 
-You do that with standard modern Python types.
+Puoi farlo usando le annotazioni di tipo standard di Python.
```
Comment (pieromacaluso) [inline docs/it/docs/index.md:301]: Suggestion: "le annotazioni ~~di tipo~~ standard"

File: docs/it/docs/index.md [hunk 3]
```diff
@@ -196,54 +191,54 @@ INFO:     Application startup complete.
...
 ![ReDoc](https://fastapi.tiangolo.com/img/index/index-02-redoc-simple.png)
 
-## Example upgrade
+## Esempio più avanzato
 
-Now modify the file `main.py` to receive a body from a `PUT` request.
+Adesso modifica il file `main.py` per ricevere un _body_ da una richiesta `PUT`.
 
-Declare the body using standard Python types, thanks to Pydantic.
+Dichiara il _body_ usando le annotazioni di tipo standard di Python, grazie a Pydantic.
```
Comment (pieromacaluso) [inline docs/it/docs/index.md:241]: Suggestion: "le annotazioni ~~di tipo~~ standard"

File: docs/it/docs/index.md [hunk 4]
```diff
@@ -25,29 +21,28 @@
...
 
-**Source Code**: <a href="https://github.com/tiangolo/fastapi" target="_blank">https://github.com/tiangolo/fastapi</a>
+**Codice Sorgente**: <a href="https://github.com/tiangolo/fastapi" target="_blank">https://github.com/tiangolo/fastapi</a>
 
 ---
 
-FastAPI is a modern, fast (high-performance), web framework for building APIs with Python 3.6+ based on standard Python type hints.
-
-The key features are:
+FastAPI è un web framework moderno e veloce (ad alte prestazioni) per progettare API con Python 3.6+, basato sulle annotazioni di tipo standard di Python.
```
Comment (pieromacaluso) [inline docs/it/docs/index.md:30]: Suggestion: "le annotazioni ~~di tipo~~ standard"

File: docs/it/docs/index.md [hunk 5]
```diff
@@ -273,175 +268,175 @@ def update_item(item_id: int, item: Item):
...
+### Ricapitolando
 
-In summary, you declare **once** the types of parameters, body, etc. as function parameters. 
+Ricapitolando, è sufficiente dichiarare **una sola volta** i tipi dei parametri, del body, ecc. come parametri di funzioni.
 
-You do that with standard modern Python types.
+Puoi farlo usando le annotazioni di tipo standard di Python.
 
-You don't have to learn a new syntax, the methods or classes of a specific library, etc.
+Non c'è bisogno di imparare una nuova sintassi, metodi o classi specifici a una libreria, ecc.
```
Comment (pieromacaluso) [inline docs/it/docs/index.md:303]: "Non c'è bisogno di imparare nuovi metodi, sintassi o classi di una specifica libreria, ecc."
[reference](https://www.impariamoinsieme.com/aggettivo-e-concordanza-con-il-nome/#:~:text=Gli%20aggettivi%20concordano%20sempre%20nel,pi%C3%B9%20il%20discorso%20%C3%A8%20diverso.&text=se%20i%20nomi%20sono%20tutti,concorda%20al%20femminile%20plurale.)

File: docs/it/docs/index.md [hunk 6]
```diff
@@ -273,175 +268,175 @@ def update_item(item_id: int, item: Item):
...
+Tornando al precedente esempio, **FastAPI**:
+
+* Validerà che c'è un `item_id` nel percorso delle richieste `GET` e `PUT`.
+* Validerà che `item_id` è di tipo `int` per le richieste `GET` e `PUT`.
+    * Se non lo è, il client vedrà un errore chiaro e utile.
+* Controllerà se c'è un parametro opzionale chiamato `q` (per esempio `http://127.0.0.1:8000/items/foo?q=somequery`) per le richieste `GET`.
+    * Siccome il parametro `q` è dichiarato con `= None`, è opzionale.
+    * Senza il `None` sarebbe stato obbligatorio specificarlo (come lo è il corpo della richiesta `PUT`).
+* Per le richieste `PUT` su `/items/{item_id}`, leggerà il corpo come JSON e
+    * Controlle che abbia un attributo obbligatorio `name`, e che sia una `str`.
```
Comment (pieromacaluso) [inline docs/it/docs/index.md:356]: "Controlla" or "Controllerà"

**Review by PieroMacaluso — APPROVED** [review]
I inserted just a few typos and comments on the translation. After these quick fixes, it could be approved.

**Comment by csr** [comment]
Thank you so much @pieromacaluso! I'll try to get this done asap.

File: docs/it/docs/index.md [hunk 7]
```diff
@@ -1,12 +1,8 @@
-
-{!../../../docs/missing-translation.md!}
-
-
 <p align="center">
   <a href="https://fastapi.tiangolo.com"><img src="https://fastapi.tiangolo.com/img/logo-margin/logo-teal.png" alt="FastAPI"></a>
 </p>
 <p align="center">
-    <em>FastAPI framework, high performance, easy to learn, fast to code, ready for production</em>
+    <em>Framework FastAPI, alte prestazioni, facile da imparare, veloce da programmare, pronto per il rilascio in produzione</em>
```
Comment (rpalloni) [inline docs/it/docs/index.md:5]: <em>Framework FastAPI, elevate prestazioni, facile da imparare, veloce da programmare, pronto per il rilascio in produzione</em>

Comment (rpalloni) [inline docs/it/docs/index.md:5]: <em>Framework FastAPI, prestazioni elevate, facile da imparare, veloce da programmare, pronto per il rilascio in produzione</em>

File: docs/it/docs/index.md (hunk 4 above)
Comment (rpalloni) [inline docs/it/docs/index.md:30]: (con prestazioni elevate)

File: docs/it/docs/index.md [hunk 8]
```diff
@@ -25,29 +21,28 @@
...
-* **Standards-based**: Based on (and fully compatible with) the open standards for APIs: <a href="https://github.com/OAI/OpenAPI-Specification" class="external-link" target="_blank">OpenAPI</a> (previously known as Swagger) and <a href="http://json-schema.org/" class="external-link" target="_blank">JSON Schema</a>.
+* **Veloce**: Prestazioni elevate, alla pari di **NodeJS** e **Go** (grazie a Starlette e Pydantic). [Uno dei framework in Python più veloci in circolazione](#performance).
+* **Veloce da programmare**: Velocizza il lavoro consentendoti di rilasciare nuove funzionalità tra il 200% e il 300% più velocemente. *
+* **Meno errori**: Riduce di circa il 40% gli erorri commessi dagli sviluppatori. *
+* **Intuitivo**: Ottimo supporto per gli editor di testo con <abbr title="anche noto come IntelliSense">autocompletamento</abbr> ovunque. Perdi meno tempo a trovare errori.
+* **Facile**: Progettato per essere facile da usare e imparare. Perdi meno tempo a consultare la documentazione.
+* **Breve**: Minimizza la duplicazione di codice. Multiple funzionalità con ciascuna dichiarazione di parametri. Meno errori.
```
Comment (rpalloni) [inline docs/it/docs/index.md:39]: * **Sintetico**

File: docs/it/docs/index.md [hunk 9]
```diff
@@ -171,15 +166,15 @@ async def read_item(item_id: int, q: str = None):
     return {"item_id": item_id, "q": q}
 ```
 
-**Note**:
+**Nota**:
 
-If you don't know, check the _"In a hurry?"_ section about <a href="https://fastapi.tiangolo.com/async/#in-a-hurry" target="_blank">`async` and `await` in the docs</a>.
+Se vuoi approfondire, consulta la sezione _"In a hurry?"_ su <a href="https://fastapi.tiangolo.com/async/#in-a-hurry" target="_blank">`async` e `await` nella documentazione</a>.
```
Comment (rpalloni) [inline docs/it/docs/index.md:171]: _"Di fretta?"_

**Review by rpalloni — CHANGES_REQUESTED** [review]
IT docs suggestions

File: docs/it/docs/index.md [hunk 10]
```diff
@@ -25,29 +21,28 @@
...
-* **Fewer bugs**: Reduce about 40% of human (developer) induced errors. *
-* **Intuitive**: Great editor support. <abbr title="also known as auto-complete, autocompletion, IntelliSense">Completion</abbr> everywhere. Less time debugging.
-* **Easy**: Designed to be easy to use and learn. Less time reading docs.
-* **Short**: Minimize code duplication. Multiple features from each parameter declaration. Fewer bugs.
-* **Robust**: Get production-ready code. With automatic interactive documentation.
-* **Standards-based**: Based on (and fully compatible with) the open standards for APIs: <a href="https://github.com/OAI/OpenAPI-Specification" class="external-link" target="_blank">OpenAPI</a> (previously known as Swagger) and <a href="http://json-schema.org/" class="external-link" target="_blank">JSON Schema</a>.
+* **Veloce**: Prestazioni elevate, alla pari di **NodeJS** e **Go** (grazie a Starlette e Pydantic). [Uno dei framework in Python più veloci in circolazione](#performance).
+* **Veloce da programmare**: Velocizza il lavoro consentendoti di rilasciare nuove funzionalità tra il 200% e il 300% più velocemente. *
```
Comment (bsab) [inline docs/it/docs/index.md:35]:
```suggestion
* **Veloce da programmare**: Velocizza il lavoro consentendo il rilascio di nuove funzionalità tra il 200% e il 300% in maniera più rapida. *
```

File: docs/it/docs/index.md [hunk 11]
```diff
@@ -25,29 +21,28 @@
...
+* **Meno errori**: Riduce di circa il 40% gli erorri commessi dagli sviluppatori. *
+* **Intuitivo**: Ottimo supporto per gli editor di testo con <abbr title="anche noto come IntelliSense">autocompletamento</abbr> ovunque. Perdi meno tempo a trovare errori.
+* **Facile**: Progettato per essere facile da usare e imparare. Perdi meno tempo a consultare la documentazione.
+* **Breve**: Minimizza la duplicazione di codice. Multiple funzionalità con ciascuna dichiarazione di parametri. Meno errori.
+* **Robusto**: Scrivi codice pronto per essere rilasciato in produzione con una documentazione dell'API interattiva generata automaticamente.
```
Comment (bsab) [inline docs/it/docs/index.md:40]:
```suggestion
* **Robusto**: Scrittura di codice pronto per essere rilasciato in produzione con una documentazione dell'API interattiva generata automaticamente.
```

File: docs/it/docs/index.md [hunk 12]
```diff
@@ -25,29 +21,28 @@
...
+* **Basato sugli standard**: Pienamente compatibile con gli open standard per le API: <a href="https://github.com/OAI/OpenAPI-Specification" class="external-link" target="_blank">OpenAPI</a> (prima conosciuto come Swagger) e <a href="http://json-schema.org/" class="external-link" target="_blank">JSON Schema</a>.
 
-<small>* estimation based on tests on an internal development team, building production applications.</small>
+<small>* Questa stima è basata su test eseguiti all'interno di un team di sviluppo di applicazioni professionali.</small>
```
Comment (bsab) [inline docs/it/docs/index.md:43]:
```suggestion
<small>* Stima basata sull'esito di test eseguiti da un team di sviluppo per applicazioni professionali.</small>
```

File: docs/it/docs/index.md [hunk 13]
```diff
@@ -116,7 +111,7 @@ $ pip install fastapi
 
 </div>
 
-You will also need an ASGI server, for production such as <a href="http://www.uvicorn.org" class="external-link" target="_blank">Uvicorn</a> or <a href="https://gitlab.com/pgjones/hypercorn" class="external-link" target="_blank">Hypercorn</a>.
+Avrai anche bisogno di un server ASGI per il rilascio in produzione come <a href="http://www.uvicorn.org" class="external-link" target="_blank">Uvicorn</a> oppure <a href="https://gitlab.com/pgjones/hypercorn" class="external-link" target="_blank">Hypercorn</a>.
```
Comment (bsab) [inline docs/it/docs/index.md:114]:
```suggestion
Per il rilascio in produzione, sarà necessario un server ASGI come <a href="http://www.uvicorn.org" class="external-link" target="_blank">Uvicorn</a> oppure <a href="https://gitlab.com/pgjones/hypercorn" class="external-link" target="_blank">Hypercorn</a>.
```

File: docs/it/docs/index.md [hunk 14]
```diff
@@ -196,54 +191,54 @@ INFO:     Application startup complete.
...
+<summary>Informazioni sul comando <code>uvicorn main:app --reload</code>...</summary>
 
-The command `uvicorn main:app` refers to:
+Vediamo il comando `uvicorn main:app` in dettaglio:
 
-* `main`: the file `main.py` (the Python "module").
-* `app`: the object created inside of `main.py` with the line `app = FastAPI()`.
-* `--reload`: make the server restart after code changes. Only do this for development.
+* `main`: il `main.py` (il "modulo" Python).
+* `app`: l'oggetto creato dentro `main.py` con la riga di codice `app = FastAPI()`.
```
Comment (bsab) [inline docs/it/docs/index.md:199]:
```suggestion
* `app`: l'oggetto istanziato all'interno del `main.py` tramite l'istruzione `app = FastAPI()`.
```

File: docs/it/docs/index.md [hunk 15]
```diff
@@ -196,54 +191,54 @@ INFO:     Application startup complete.
...
 ```
 
-You already created an API that:
+Hai appena creato un'API che:
 
-* Receives HTTP requests in the _paths_ `/` and `/items/{item_id}`.
-* Both _paths_ take `GET` <em>operations</em> (also known as HTTP _methods_).
-* The _path_ `/items/{item_id}` has a _path parameter_ `item_id` that should be an `int`.
-* The _path_ `/items/{item_id}` has an optional `str` _query parameter_ `q`.
+* Riceve richieste HTTP ai _paths_ `/` e `/items/{item_id}`.
```
Comment (bsab) [inline docs/it/docs/index.md:216]:
```suggestion
* Riceve richieste HTTP sui _paths_ `/` e `/items/{item_id}`.
```

File: docs/it/docs/index.md [hunk 16]
```diff
@@ -196,54 +191,54 @@ INFO:     Application startup complete.
...
+* Entrambi i _paths_ ricevono `GET` <em>operations</em> (anche chiamati <abbr title="metodi HTTP">HTTP _methods_</abbr>).
```
Comment (bsab) [inline docs/it/docs/index.md:217]:
```suggestion
* Entrambi i _paths_ accettano`GET` <em>operations</em> (conosciuti anche come <abbr title="metodi HTTP">HTTP _methods_</abbr>).
```

File: docs/it/docs/index.md [hunk 17]
```diff
@@ -273,175 +268,175 @@ def update_item(item_id: int, item: Item):
...
 
 ![Swagger UI interaction](https://fastapi.tiangolo.com/img/index/index-04-swagger-03.png)
 
-* Then click on the "Execute" button, the user interface will communicate with your API, send the parameters, get the results and show them on the screen:
+* Successivamente, fai click sul pulsante "Execute". L'interfaccia utente comunicherà con la tua API, invierà i parametri, riceverà i risultati della richiesta, e li mostrerà sullo schermo:
 
 ![Swagger UI interaction](https://fastapi.tiangolo.com/img/index/index-05-swagger-04.png)
 
-### Alternative API docs upgrade
+### Aggiornamento della documentazione alternativa
```
Comment (bsab) [inline docs/it/docs/index.md:289]: I am not convinced :/

File: docs/it/docs/index.md [hunk 18]
```diff
@@ -273,175 +268,175 @@ def update_item(item_id: int, item: Item):
...
-And now, go to <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.
+Ora vai su <a href="http://127.0.0.1:8000/redoc" class="external-link" target="_blank">http://127.0.0.1:8000/redoc</a>.
 
-* The alternative documentation will also reflect the new query parameter and body:
+* Anche la documentazione alternativa dell'API mostrerà il nuovo parametro della query e il _body_:
 
 ![ReDoc](https://fastapi.tiangolo.com/img/index/index-06-redoc-02.png)
 
-### Recap
+### Ricapitolando
```
Comment (bsab) [inline docs/it/docs/index.md:297]:
```suggestion
### Riepilogo
```

File: docs/it/docs/index.md [hunk 19]
```diff
@@ -273,175 +268,175 @@ def update_item(item_id: int, item: Item):
...
+Puoi farlo usando le annotazioni di tipo standard di Python.
 
-You don't have to learn a new syntax, the methods or classes of a specific library, etc.
+Non c'è bisogno di imparare una nuova sintassi, metodi o classi specifici a una libreria, ecc.
 
-Just standard **Python 3.6+**.
+È sufficiente usare **Python 3.6+**.
 
-For example, for an `int`:
+Per esempio, per un `int`:
```
Comment (bsab) [inline docs/it/docs/index.md:307]:
```suggestion
Per esempio, per il tipo `int`:
```

File: docs/it/docs/index.md [hunk 20]
```diff
@@ -273,175 +268,175 @@ def update_item(item_id: int, item: Item):
...
 
-...to:
+...a questa:
 
 ```Python
         ... "item_price": item.price ...
 ```
 
-...and see how your editor will auto-complete the attributes and know their types:
+...e osserva come il tuo editor di testo autocompleterà gli attributi e saprà i loro tipi:
```
Comment (bsab) [inline docs/it/docs/index.md:388]:
```suggestion
...e osserva come il tuo editor di testo autocompleterà gli attributi e sarà in grado di conoscere i loro tipi:
```

File: docs/it/docs/index.md [hunk 21]
```diff
@@ -273,175 +268,175 @@ def update_item(item_id: int, item: Item):
...
+* Convertirà da JSON a JSON automaticamente.
+* Documenterà tutto con OpenAPI, che può essere usato per:
+    * Sistemi di documentazione interattivi.
+    * Sistemi di generazione di codice dal lato client, per molti linguaggi.
+* Fornirà 2 interfacce di documentazione dell'API interattive.
 
 ---
 
-We just scratched the surface, but you already get the idea of how it all works.
+Questa è solo la punta dell'iceberg, ma dovresti avere già un'idea di come il tutto funziona.
```
Comment (bsab) [inline docs/it/docs/index.md:368]:
```suggestion
Questa è solo la punta dell'iceberg, ma si dovrebbe avere già un'idea di come il tutto funziona.
```

**Review by bsab — CHANGES_REQUESTED** [review]
I have added some comments and changes. I think the file can be merged.

**Comment by csr** [comment]
Thank you so much for contributing and for suggesting changes. I'm not actively participating in the translation efforts any longer but I would encourage you to create a separate pull request with all this good stuff.

**Pull request closed without merging**